### PR TITLE
Add DEV_WHITE_LIST param to mellanox_fw_update.yaml file

### PR DESCRIPTION
**Pull request by wmousa**

### CODE REVIEW

**Comment by wmousa** [comment]
@moshe010 Please review it

File: templates/firstboot/mellanox_fw_update.yaml [hunk 1]
```diff
@@ -144,7 +152,7 @@ resources:
                         if not line:
                             continue
                         dev = line.split()[0]
-                        if dev.endswith('.0'):
+                        if dev.endswith('.0') and (not _DEV_WHITE_LIST or dev in _DEV_WHITE_LIST):
```
Comment (moshe010) [inline templates/firstboot/mellanox_fw_update.yaml]: why _DEV_WHITE_LIST is global and not passed in the MlnxDevices init

Comment (wmousa) [reply templates/firstboot/mellanox_fw_update.yaml]: @moshe010 Done

File: templates/firstboot/mellanox_fw_update.yaml [hunk 2]
```diff
@@ -138,13 +147,14 @@ resources:
...
-                    cmd = ['lspci', '-d', '15b3:']
+                    cmd = ['lspci', '-D','-d', '15b3:']
                     out = run_command(*cmd)
                     for line in out.split('\n'):
                         if not line:
                             continue
                         dev = line.split()[0]
-                        if dev.endswith('.0'):
+                        if dev.endswith('.0') and (not self._dev_white_list or
+                                dev in self._dev_white_list):
```
Comment (moshe010) [inline templates/firstboot/mellanox_fw_update.yaml:157]: we should log a warning all the pci that are in the whitelist and not found

Comment (moshe010) [reply templates/firstboot/mellanox_fw_update.yaml:157]: Also if there are pci that ends with not zero in the whitelist we should log a warning

**Review by moshe010 — CHANGES_REQUESTED** [review]

File: templates/firstboot/mellanox_fw_update.yaml (hunk 2 above)
Comment (wmousa) [reply templates/firstboot/mellanox_fw_update.yaml:157]: > we should log a warning all the pci that are in the whitelist and not found

Done

Comment (wmousa) [reply templates/firstboot/mellanox_fw_update.yaml:157]: > Also if there are pci that ends with not zero in the whitelist we should log a warning

Done

File: templates/firstboot/mellanox_fw_update.yaml [hunk 3]
```diff
@@ -138,16 +147,21 @@ resources:
...
                             devs.append(dev)
                     self._devs = devs
                     LOG.info("Found Mellanox devices: %s", devs)
+                    other_devs = set(self._dev_white_list) - set(devs)
+                    if other_devs:
+                        LOG.warning("Uproccessed devices %s which may be not Mellanox devices, "
```
Comment (moshe010) [inline templates/firstboot/mellanox_fw_update.yaml]: s/Unproccessed/Unprocessed

File: templates/firstboot/mellanox_fw_update.yaml [hunk 4]
```diff
@@ -13,6 +13,13 @@ parameters:
     type: boolean
     default: False
     description: "Force update the fw even if it's older version"
+  DEV_WHITE_LIST:
+    type: list
+    default: []
+    description: list of MLNX devices PCIs to be processed.
+                 If the value is empty, all MLNX devices will be processed.
+                 Example, ['0000:04:00.0', '0000:81:00.0']
+                 Make sure to choose only the PCI ends with 0
```
Comment (adrianchiris) [inline templates/firstboot/mellanox_fw_update.yaml:22]: what about applying per PF configurations ? the code no only updates firmware, 
you should allow wildcard for the function.

we assume SR-IOV is not configured when this bit runs right ?

File: templates/firstboot/mellanox_fw_update.yaml (hunk 2 above)
Comment (adrianchiris) [reply templates/firstboot/mellanox_fw_update.yaml:157]: you are breaking per PF configuration here

**Review by adrianchiris — CHANGES_REQUESTED** [review]

File: templates/firstboot/mellanox_fw_update.yaml (hunk 4 above)
Comment (moshe010) [reply templates/firstboot/mellanox_fw_update.yaml]: Bt fw is per nic? So why this a problem?

Comment (moshe010) [reply templates/firstboot/mellanox_fw_update.yaml:22]: Bt fw is per nic? So why this a problem?

Comment (adrianchiris) [reply templates/firstboot/mellanox_fw_update.yaml:22]: nvconfig may be per PF

Comment (moshe010) [reply templates/firstboot/mellanox_fw_update.yaml:22]: Yes but fw is per nic, so I would to keep it all config per nic, not per pf

Comment (adrianchiris) [reply templates/firstboot/mellanox_fw_update.yaml:22]: the issue i described is not related to this commit, nvm.
before the change we only processed PF 0 as well

in NVConfig there are parameters that need to be applied per PF, however we currently dont support those in the code

Comment (moshe010) [reply templates/firstboot/mellanox_fw_update.yaml:22]: right, there is not requirement for that at least for now :)

**Comment by moshe010** [comment]
@waleedym, once you fix the typo and test it in a real deployment I will approve it.

File: templates/firstboot/mlnx_fw_update.py [hunk 5]
```diff
@@ -89,16 +91,21 @@ def discover(self):
...
             dev = line.split()[0]
-            if dev.endswith('.0'):
+            if dev.endswith('.0') and (not self._dev_white_list or
+                    dev in self._dev_white_list):
                 devs.append(dev)
         self._devs = devs
         LOG.info("Found Mellanox devices: %s", devs)
+        other_devs = set(self._dev_white_list) - set(devs)
+        if other_devs:
+             LOG.warning("Uproccessed devices %s which may be not Mellanox devices, "
```
Comment (adrianchiris) [inline templates/firstboot/mlnx_fw_update.py]: this message is unclear to me.  perhaps something like:

"Not all devices in PCI white list where discovered, these may not be mellanox devices or have their PCI function set to non zero. "?

**Comment by moshe010** [comment]
@wmousa also create PR to move the Tripleo related files to Tripleo folder

**Review by adrianchiris — COMMENTED** [review]
overall fine with the review, just a nit for the logging

**Review by adrianchiris — APPROVED** [review]

**Comment by wmousa** [comment]
@moshe010 It's ready now and tested

**Review by moshe010 — APPROVED** [review]

**Pull request merged**